logic/concludable/ConjunctionConcludable.java

```diff
@@ -112,21 +119,107 @@ public Value asValue() {
         throw GraknException.of(INVALID_CASTING, className(this.getClass()), className(Value.class));
     }
 
+    Optional<Map<Reference, Set<Reference>>> extendUnifier(
+            Variable conjVar, Variable headVar, Map<Reference, Set<Reference>> variableMapping) {
+        if (varHintsDisjoint(headVar, conjVar)) return Optional.empty();
```

Comment:
i suspect that this method should not be returning an Optional at all -- everywhere you call it you immediately do an `isPresent()` check to extract the thing inside the optional. If you just do the `varHintsDisjoint()` call as a check before `extendUnifier()`, this won't need to return an optional at all, and the caller will not need to check for optional. It'll just be `if (varHintsDisjoint()) extendUnifier(...) else return Stream.empty()`

Reply:
But then you've decoupled the failure of unifying. Essentially forcing every calling to know to do that check. I think using the Optional way is safer.

Reply:
then the name is misleading, it should be `tryExtendUnifier`